src/components/ClickCopy.vue

```diff
@@ -25,6 +25,14 @@ export default {
   props: {
     gradient: String,
     name: String,
+    vanilla: Boolean,
+    value: String,
+  },
+  mounted() {
+    this.value = this.vanilla ? getComputedStyle(this.$refs.gradientUnderline).getPropertyValue('background-image') : this.gradient
+  },
+  updated() {
+    this.value = this.vanilla ? getComputedStyle(this.$refs.gradientUnderline).getPropertyValue('background-image') : this.gradient
```

Comment:
Great use of getComputedStyle!
First I think what you want is having `value` as a `data` instead of a prop (in VueJS, props are inputs, not internal state), but maybe the best option is to have a `vanillaStyle` [computed property](https://vuejs.org/v2/guide/computed.html).

```suggestion
  },
  computed: {
    vanillaStyle() {
       return getComputedStyle(this.$refs.gradientUnderline).getPropertyValue('background-image')
    }
```

Reply:
Okay, but this doesn't solve the fact that `this.$refs.gradientUnderline` is undefined during the first render of the component. So `getComputedStyle()` breaks and the component isn't rendered.

The problem is that `v-clipboard:copy` evaluates the value during the first render, so I can't use the `ref` yet. I don't know if there is another way to do it.

Doing without the `ref` is hard (with `document` DOM methods) because it's hard to find the good query.

Maybe the only solution is to move the logic up in the parent component (i.e. `Gradient`)...

Reply:
What do you think of:
```js
if (!this.refs.gradientUnderline) {
  return null
}
return ...
```

Reply:
Then it won't copy anything and you'll need to select other directions to re-render the component so as to have something to copy... It's not a solution I think 😁

Reply:
Damn it...

Reply:
I found it, `refs` is not reactive (https://stackoverflow.com/a/43531779), but, `v-clipboard:copy` can have a computed value in it.

Here a code that works:

```js
  data() {
    return {
      mounted: false,
    };
  },
  props: {
    gradient: String,
    name: String,
    vanilla: Boolean,
  },
  mounted() {
    this.mounted = true;
  },
  computed: {
    value() {
      if (!this.mounted) {
        return null;
      }
      if (this.vanilla) {
        return getComputedStyle(this.$refs.gradientUnderline).getPropertyValue(
          "background-image"
        );
      }
      return this.gradient;
    },
  },
```

Reply:
Okay, thanks. I just implemented it and it works well!